apps/settings/js/media_storage.js

```diff
@@ -430,78 +431,46 @@ require([
 
   Volume.prototype.unmountSDCard = function volume_unmountSDCard(evt) {
     // Pop up a confirm window before unmount SD card.
-    var popup = document.getElementById('unmount-sdcard-dialog');
-    var cancelBtn = document.getElementById('unmount-sdcard-cancel-btn');
-    var okBtn = document.getElementById('unmount-sdcard-ok-btn');
-
     var self = this;
-    var confirmHandler = function() {
-      enablePopup(false);
-      // Unmount SD card
-      self.storages.sdcard.unmount();
-      self.isUnmounting = true;
-    };
-
-    var cancelHandler = function() {
-      enablePopup(false);
-    };
 
-    var enablePopup = function Vf_enablePopup(enabled) {
-      if (enabled) {
-        okBtn.addEventListener('click', confirmHandler);
-        cancelBtn.addEventListener('click', cancelHandler);
-        popup.hidden = false;
-      } else {
-        okBtn.removeEventListener('click', confirmHandler);
-        cancelBtn.removeEventListener('click', cancelHandler);
-        popup.hidden = true;
+    DialogService.confirm('unmount-sdcard-confirmation', {
+      submitButton: 'ok',
+      cancelButton: 'cancel'
+    }).then(function(result) {
+      if (result.type === 'submit') {
+        // Unmount SD card
+        self.storages.sdcard.unmount();
+        self.isUnmounting = true;
       }
-    };
-    enablePopup(true);
```

Comment:
haha, if not this question, I won't have the motivation to take a look at the mechanism of DialogService.
I'll fix the rest of the part later and send u a review request again. 

Thanks :)